File: matematica/python/1307.py
# -*- coding:utf-8 -*-
def maior(num1, num2):
    if num1 >= num2:
        maior = num1
        menor = num2
    else:
        maior = num2
        menor = num1
    return maior, menor
    
def calcula_mdc(dec1, dec2):
    if dec1 != dec2:
        mdc = 1
        maior_num, menor_num = maior(dec1, dec2)
        dec1 = maior_num
        dec2 = menor_num
        while (dec2 != 0):
            dec1, dec2 = menor_num, maior_num % dec2
            menor_num = dec2
            maior_num = dec1
        return maior_num
    else:
        return dec1
    
N = int(input())
for i in range(N):
    string1 = int(input(), 2)
    string2 = int(input(), 2)
    tudo_amor = calcula_mdc(string1, string2)
    if tudo_amor > 1:
        print("Pair #%d: All you need is love!"%(i+1))
    elif tudo_amor == 1:
        print("Pair #%d: Love is not all you need!"%(i+1))

File: matematica/c++/1307.cpp
#include <iostream>
#define endl '\n';
typedef unsigned long long ull;

using namespace std;

ull mdc(ull x, ull y){
    return (y == 0? x: mdc(y, x%y));
}
 
int main() {
    int N;
    string s1, s2;
    ull n1, n2;
    
    cin >> N;
    
    for (int i=1; i<=N; i++){
        cin >> s1;
        cin >> s2;
        n1 = stoul(s1, NULL, 2);
        n2 = stoul(s2, NULL, 2);
        
        cout<< "Pair #"<< i;
        if (mdc(n1, n2) != 1){
            cout<< ": All you need is love!"<< endl;
        }
        else{
            cout<< ": Love is not all you need!"<< endl;
        }
        
    }
 
    return 0;
}